icaruscode/Filters/FilterCRTPMTMatching_module.cc

```diff
@@ -13,556 +13,607 @@
-    int                        fRun;    ///< number of the run being processed.
-    int                        fSubRun; ///< number of the sub-run being processed.
-
-    // add trigger data product vars
-    unsigned int               m_gate_type;
-    std::string                m_gate_name;
-    uint64_t                   m_trigger_timestamp;
-    uint64_t                   m_gate_start_timestamp;
-    uint64_t                   m_trigger_gate_diff;
-    uint64_t		       m_gate_width;
-
-    std::string		       fFilterLevel;		// Filter level, default is loose 
-    double		       fOpHitAmplitude;    	// ADC threshold for the PMT
-    int                        fnOpHitToTrigger;        // Number of OpHit above threshold to mimic the trigger
-    double 		       fTimeOfFlightInterval;   // Time of Flight interval to find the match
-    bool                       fOutputTree;             // Output tree or not
-    map<int, art::InputTag>    fFlashLabels;
-
-    TTree*                     fMatchTree;
-
-    // matchTree vars
-    vector<double>             fOpFlashPos;             ///< Position of the optical Flash Barycenter.
-    double                     fOpFlashTime;            ///< Time of the optical Flash w.r.t. Global Trigger.
-    bool                       fInTime;                 ///< Was the OpFlash in beam spill time?
-    double                     fOpFlashPE;              ///< Total PEs of the optical Flash.
-    vector<double>             fOpHitX;                 ///< X position of the OP hit.
-    vector<double>             fOpHitY;                 ///< Y position of the OP hit.
-    vector<double>             fOpHitZ;                 ///< Z position of the OP hit.
-    vector<double>             fOpHitT;                 ///< T of the OP hit.
-    vector<double>             fOpHitA;                 ///< Amplitude (ADC) of the OP hit.
-    int                        fNCRTmatch;
-    vector<vector<double>>     fMatchedCRTpos;          ///< Position of the matched CRTs.
-    vector<double>             fMatchedCRTtime;         ///< Time of the matched CRT Hits w.r.t. Global Trigger.
-    vector<int>                fMatchedCRTregion;       ///< Region of the matched CRT Hits.
-    vector<vector<int>>        fMatchedCRTmodID;        ///< ModID of the CRT module.
-    vector<int>                fMatchedCRTsys;          ///< Subsystem of the matched CRT Hit.
-    vector<double>             fMatchedCRTamplitude;    ///< Amplitude of the matched CRT Hit.
-    vector<double>             fTofOpFlash;             ///< Time of Flight between matched CRT and Optical Flash.
-    vector<double>             fCRTGateDiff;            ///< Difference between CRTHit and BeamGate opening.
-    int                        fEventType;              ///< Was triggered the event?
-
-    geo::GeometryCore const*   fGeometryService; ///< pointer to Geometry provider.
+class icarus::crt::FilterCRTPMTMatching : public art::EDFilter {
```

Comment:
It would be good to have a long and exhaustive description of what the module does and how to configure it, in particular for the different levels. A reference to a SBN DocDB document is also welcome.
```suggestion
/**
 * @brief Rejects events with only incoming particles in time with the beam.
 *
 * Here is a long and detailed explanation on what are the choices.
 */
class icarus::crt::FilterCRTPMTMatching : public art::EDFilter {
```